test/extended/apiserver/latency.go

```diff
@@ -0,0 +1,101 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	promhelper "github.com/openshift/origin/test/extended/util/prometheus"
+)
+
+// TODO: remove these variables once we move to route instead of exec pod
+var (
+	oc                   *exutil.CLI
+	url, bearerToken, ns string
+	execPod              *v1.Pod
+)
+
+var _ = g.Describe("[sig-instrumentation][Late] API Server latency", func() {
+	defer g.GinkgoRecover()
+
+	oc = exutil.NewCLIWithoutNamespace("apiserver-latency")
+
+	g.BeforeEach(func() {
+		err := exutil.WaitForAnImageStream(
+			oc.AdminImageClient().ImageV1().ImageStreams("openshift"), "tools",
+			exutil.CheckImageStreamLatestTagPopulated, exutil.CheckImageStreamTagNotFound,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var ok bool
+		url, _, bearerToken, ok = promhelper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	// Lantecy thresholds collected from:
```

Comment:
Can you specify the query you used?  I think that will give me enough, when combined with these links, that I could rebuild what you got.

Reply:
make sense